src/interfaces/MutatorFunction.ts

```diff
@@ -1,4 +1,4 @@
 import ActionMessage from './ActionMessage';
 
-type MutatorFunction<T extends ActionMessage> = (actionMessage: T) => void;
+type MutatorFunction<T extends ActionMessage, U> = void extends U ? (actionMessage: T) => U : never;
```

Comment:
>void extends U [](start = 51, length = 14)

Can you add some comments here about what this actually does?  I guess that it evaluates to false if U is `void` or true if U is anything else?